R/input-time-series.R

```diff
@@ -565,3 +567,167 @@ build_hierarchy_label <- function(meta_areas) {
   labels[is.na(labels)] <- root_name
   labels
 }
+
+##' Compare aggregated district ART inputs + spectrum totals
+
+##' Compare aggregated subnational ART inputs + spectrum totals for comparison table
+##'
+##' @param art Path to file containing ART data or ART data object
+##' @param shape Path to file containing geojson areas data or areas data object
+##' @param pjnz Path to zip file containing spectrum pjnz file/s
+##' @keywords internal
+prepare_art_spectrum_comparison <- function(art, shape, pjnz) {
+
+
+  ## Check if shape is object or file path
+  if(!inherits(shape, "sf")) {
+    shape <- read_area_merged(shape) }
+
+  ## Check if art is object or file path
+  if(!inherits(art, c("spec_tbl_df","tbl_df","tbl","data.frame" ))) {
+    art <- read_art_number(art, all_columns = TRUE)}
+
+  ## PJNZ either object or file path
+  if (!inherits(pjnz, "spec_program_data")) {
+    pjnz <- extract_pjnz_program_data(pjnz) }
```

Comment:
Sorry this is pedantic

```suggestion
    shape <- read_area_merged(shape)
  }

  ## Check if art is object or file path
  if(!inherits(art, c("spec_tbl_df","tbl_df","tbl","data.frame" ))) {
    art <- read_art_number(art, all_columns = TRUE)
  }

  ## PJNZ either object or file path
  if (!inherits(pjnz, "spec_program_data")) {
    pjnz <- extract_pjnz_program_data(pjnz)
  }
```